pkg/build/strategies/layered/layered.go

```diff
@@ -63,14 +64,24 @@ func (b *Layered) CreateDockerfile(config *api.Config) error {
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", b.config.BuilderImage))
-	buffer.WriteString(fmt.Sprintf("COPY scripts %s\n", locations[0]))
+	// if -s / --scripts / --scripts-url are not specified, there will not be a scripts subdir
+	// off of the working dir, so don't copy here, and we'll have to tweak the scripts URL to use the env var
+	// STI_SCRIPTS_PATH
+	scriptsIncluded := len(config.ScriptsURL) > 0
```

Comment:
for (3) i'm not talking about a default value in the label, i'm talking about a case where the user explicitly invokes s2i with a "--scripts-url image://some/path" argument, which means they are specifying "use these external scripts" (case 1) but the scripts are really in the image, not downloaded from some external location, meaning you stlil still not have a "scripts" directory in the tar.

so it's a hybrid of case 1+3.  you can't rely on scriptsurl being len==0 because it will be non-zero but there will still be no scripts directory.  (I think.  i'm really just raising this as a "something to doublecheck", i'm not 100% certain on the behavior)